frontend/src/js/controllers/challengeCtrl.js

```diff
@@ -2008,6 +2008,63 @@
             });
         };
 
+        // Edit Challenge Start and End Date
+        vm.challengeDateDialog = function(ev) {
+            vm.challengeStartDate = vm.page.start_date;
+            vm.challengeStartDate = moment(vm.challengeStartDate);
+            vm.challengeEndDate = vm.page.end_date;
+            vm.challengeEndDate = moment(vm.challengeEndDate);
+            $mdDialog.show({
+                scope: $scope,
+                preserveScope: true,
+                targetEvent: ev,
+                templateUrl: 'dist/views/web/challenge/edit-challenge/edit-challenge-date.html',
+                escapeToClose: false
+            });
+        };
+
+        vm.editChallengeDate = function(editChallengeDateForm) {
+            if (editChallengeDateForm) {
+                var challengeHostList = utilities.getData("challengeCreator");
+                for (var challenge in challengeHostList) {
+                    if (challenge == vm.challengeId) {
+                        vm.challengeHostId = challengeHostList[challenge];
+                        break;
+                    }
+                }
+                parameters.url = "challenges/challenge_host_team/" + vm.challengeHostId + "/challenge/" + vm.challengeId;
+                parameters.method = 'PATCH';
+                if (new Date(vm.page.start_date).valueOf() < new Date(vm.page.end_date).valueOf()) {
+                    parameters.data = {
+                        "start_date": vm.challengeStartDate,
+                        "end_date": vm.challengeEndDate
+                    };
+                    parameters.callback = {
+                        onSuccess: function(response) {
+                            var status = response.status;
+                            utilities.hideLoader();
+                            if (status === 200) {
+                                $mdDialog.hide();
```

Comment:
you can edit the date field here and update them with the new date in the callback.

Reply:
This is really eye-opening, thanks!